manifests/config.pp

```diff
@@ -1,7 +1,19 @@
 # == Class chocolatey::config
-#
-# This class is called from chocolatey for service config.
-#
-# NOTE: CURRENTLY NON-FUNCTIONAL
 class chocolatey::config {
+  assert_private()
+
+  #todo: check choco version from custom choco_version fact
+
+  $_choco_exe_path = "${chocolatey::choco_install_location}\\bin\\choco.exe"
+
+  $_enable_autouninstaller = $chocolatey::enable_autouninstaller ? {
+    false => 'disable',
+    default => 'enable'
+  }
+
+  exec { "chocolatey_autouninstaller_${_enable_autouninstaller}":
+    path    => $::path,
+    command => "${_choco_exe_path} feature -r ${_enable_autouninstaller} -n autoUninstaller",
+    unless  => "${_choco_exe_path} feature list -r | findstr /X /I \"autoUninstaller.*\\[${_enable_autouninstaller}d\\]\"",
```

Comment:
This line when run against a regular command line pipes correctly.
![image](https://cloud.githubusercontent.com/assets/63502/9510262/40447864-4c2f-11e5-9b67-c084ae323357.png)


When run in Puppet, it pushes it all together to be part of the chocolatey call instead of letting the commands split.

![image](https://cloud.githubusercontent.com/assets/63502/9510320/fe7c4546-4c2f-11e5-9ff7-33756bce26c2.png)

~~~
Debug: Exec[chocolatey_autouninstaller_enable](provider=windows): Executing chec
k 'C:\ProgramData\chocolatey\bin\choco.exe feature list -r | findstr /X /I "auto
Uninstaller.*\[enabled\]"'
Debug: Executing 'C:\ProgramData\chocolatey\bin\choco.exe feature list -r | find
str /X /I "autoUninstaller.*\[enabled\]"'
Debug: /Stage[main]/Chocolatey::Config/Exec[chocolatey_autouninstaller_enable]/u
nless: A single features command must be listed. Please see the help menu for th
ose commands
Debug: Exec[chocolatey_autouninstaller_enable](provider=windows): Executing 'C:\
ProgramData\chocolatey\bin\choco.exe feature -r enable -n autoUninstaller'
Debug: Executing 'C:\ProgramData\chocolatey\bin\choco.exe feature -r enable -n a
utoUninstaller'
Notice: /Stage[main]/Chocolatey::Config/Exec[chocolatey_autouninstaller_enable]/
returns: executed successfully
~~~

The following command will process the same bad output: `C:\ProgramData\chocolatey\bin\choco.exe feature list -r bob`

![image](https://cloud.githubusercontent.com/assets/63502/9510314/d44feda4-4c2f-11e5-9aa4-bf05852175a5.png)